src/openforms/registrations/contrib/microsoft_graph/plugin.py

```diff
@@ -12,18 +14,42 @@
 )
 from openforms.submissions.models import Submission, SubmissionReport
 from openforms.submissions.tasks.registration import set_submission_reference
+from openforms.template import render_from_string
 
 from ...base import BasePlugin
 from ...exceptions import NoSubmissionReference, RegistrationFailed
 from ...registry import register
+from .config import MicrosoftGraphOptionsSerializer
 
 
 @register("microsoft-graph")
 class MSGraphRegistration(BasePlugin):
     verbose_name = _("Microsoft Graph (OneDrive/SharePoint)")
+    configuration_options = MicrosoftGraphOptionsSerializer
+
+    def _get_folder_name(self, submission: Submission, options: dict) -> "Path":
+        date = timezone.now().date()
+        folder_path = render_from_string(
+            options["folder_path"],
+            {
+                "year": "{date:%Y}".format(date=date),
+                "month": "{date:%m}".format(date=date),
+                "day": "{date:%d}".format(date=date),
+            },
+        )
+
+        return (
+            Path(folder_path)
+            / Path(slugify(submission.form.admin_name))
+            / Path(submission.public_registration_reference)
+        )
 
-    def _get_folder_name(self, submission: Submission):
-        return f"open-forms/{slugify(submission.form.admin_name)}/{submission.public_registration_reference}"
+    def _get_filename(self, *args: Union[str, "Path"]) -> "Path":
+        filename = Path("/")
+        for arg in args:
+            part = Path(arg) if not isinstance(arg, Path) else arg
+            filename = filename / part
```

Comment:
```suggestion
        filename = Path("/", *args)
```

`Path` takes *args too :-)

```py
>>> Path("/", "foo", "bar")
PosixPath('/foo/bar')
```